AIMSICD/src/main/java/com/secupwn/aimsicd/smsdetection/AdvancedUserActivity.java

```diff
@@ -49,48 +46,32 @@
 
     @InjectView(R.id.spinner)
     private Spinner spinner;
-    private List<AdvanceUserItems> msgItems;
+
+    private Realm realm;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        dbAccess = new AIMSICDDbAdapter(getApplicationContext());
 
-        try {
-            msgItems = dbAccess.getDetectionStrings();
-        } catch (Exception ee) {
-            log.error("Database error", ee);
-            msgItems = new ArrayList<>();
-            AdvanceUserItems advUserItems = new AdvanceUserItems();
-            advUserItems.setDetection_string("NO DATA");
-            advUserItems.setDetection_type("No TYPE");
-            msgItems.add(advUserItems);
-        }
+        realm = Realm.getDefaultInstance();
+
+        RealmResults<SmsDetectionString> msgItems = realm.allObjects(SmsDetectionString.class);
```

Comment:
Note that `allObjects` isn't sorted, you probably want to use `allObjectsSorted` or `where(SmsDetectionString.class).findAllSorted()`.